clouddriver-kubernetes-v2/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2CredentialsTest.java

```diff
@@ -0,0 +1,376 @@
+import com.netflix.spectator.api.ManualClock;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Tag;
+import com.netflix.spectator.api.Timer;
+import com.netflix.spinnaker.clouddriver.kubernetes.config.KubernetesConfigurationProperties.ManagedAccount;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.AccountResourcePropertyRegistry;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.GlobalResourcePropertyRegistry;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesSpinnakerKindMap;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesManifest;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.names.KubernetesManifestNamer;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.handler.KubernetesUnregisteredCustomResourceHandler;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.job.KubectlJobExecutor;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.job.KubectlJobExecutor.KubectlException;
+import com.netflix.spinnaker.clouddriver.names.NamerRegistry;
+import com.netflix.spinnaker.clouddriver.security.ProviderVersion;
+import com.netflix.spinnaker.kork.configserver.CloudConfigResourceService;
+import com.netflix.spinnaker.kork.configserver.ConfigFileService;
+import java.util.HashMap;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+@RunWith(JUnitPlatform.class)
+final class KubernetesV2CredentialsTest {
+  private static final String ACCOUNT_NAME = "my-account";
+  private static final String DEPLOYMENT_NAME = "my-deployment";
+  private static final String NAMESPACE = "my-namespace";
+
+  private KubernetesV2Credentials getCredentials(
+      Registry registry, KubectlJobExecutor jobExecutor) {
+    KubernetesV2Credentials.Factory factory =
+        new KubernetesV2Credentials.Factory(
+            registry,
+            new NamerRegistry(ImmutableList.of(new KubernetesManifestNamer())),
+            jobExecutor,
+            new ConfigFileService(new CloudConfigResourceService()),
+            new AccountResourcePropertyRegistry.Factory(
+                new GlobalResourcePropertyRegistry(
+                    ImmutableList.of(), new KubernetesUnregisteredCustomResourceHandler())),
+            new KubernetesKindRegistry.Factory(
+                new GlobalKubernetesKindRegistry(ImmutableList.of())),
+            new KubernetesSpinnakerKindMap(ImmutableList.of()));
+    ManagedAccount managedAccount = new ManagedAccount();
+    managedAccount.setName("my-account");
+    managedAccount.setProviderVersion(ProviderVersion.v2);
+    return factory.build(managedAccount);
+  }
+
+  private KubernetesManifest getManifest() {
+    KubernetesManifest manifest = new KubernetesManifest();
+    manifest.put("metadata", new HashMap<>());
```

Comment:
> This commit also updates the constructor of KubernetesManifest to
set the metadata to an empty map.

Did you end up deciding not to do this and just manually set it to an empty map here?

Reply:
Yes, good catch!  It turns out setting it in the constructor broke deserialization so I removed that change (but not the comment).